File: Tanque de Nível/modelo.py
from matplotlib.pyplot import *
from numpy import *

A = 10.0             # area do tanque (m**2)
H = [100.0]           # altura inicial de fluido no tanque (m)
a = 0.0              # intervalos de integracao
b = 10.0              # 
N = 1000              # discretizacao do intervalo de integracao
passo = (b-a) / N    # passo
t = linspace(\
             a,\
             b,\
             N,\
             endpoint=False\
            ) # vetor de tempo

fQe = lambda t: 10*t # razao da vazao de entrada (m**3)
fQs = lambda t: 50*pi*cos(t) # razao de vazao de saida   (m**3)

Qe = fQe(t)         
Qs = fQs(t)

for i in range(N-1):
    H.append(H[i] + passo*(Qe[i] - Qs[i])/A)  # altura em função dos parametros de entrada

 
plot(t.tolist(),H)
xlabel('Tempo (s)')
ylabel('Altura de fluido no tanque (m)')
grid()
show()

File: Tanque de Nível/modelo.m
clc, clear all, close all

A = 10;            % area do tanque em m^2
H(1) = 30;         % altura inicial da coluna de liquido no tanque
a = 0;             % ponto inicial usado no metodo de euler
b = 5;             % ponto final usado no metodo de euler
N = 100;           % numero de pontos usado no metodo de euler
passo = (b-a) / N; % passo usado no método de euler
t = passo.*(0:1:N) % tempo de 0 a 5 segundos
Qi = 20.*t;        % vazão de entrada m^3/s
Qo = 10.*t;        % vazão de saída  m^3/s
 
for i = 1:N
    H(i+1) = H(i) + passo.*(Qi(i) - Qo(i))./ A ; % altura em função dos parametros de entrada
end
 
plot(t,H);
xlabel('Tempo (s)');
ylabel('Nivel de fluido no tanque(m)');
grid();
sleep(3);